tests/pairedomics/test_strain_mappings_generator.py

```diff
@@ -0,0 +1,288 @@
+import json
+from nplinker.pairedomics import extract_mappings_ms_filename_spectrum_id
+from nplinker.pairedomics import \
+    extract_mappings_original_genome_id_resolved_genome_id
+from nplinker.pairedomics import extract_mappings_resolved_genome_id_bgc_id
+from nplinker.pairedomics import extract_mappings_strain_id_ms_filename
+from nplinker.pairedomics import extract_mappings_strain_id_original_genome_id
+from nplinker.pairedomics import get_mappings_strain_id_bgc_id
+from nplinker.pairedomics import get_mappings_strain_id_spectrum_id
+from nplinker.pairedomics import podp_generate_strain_mappings
+from nplinker.strain_collection import StrainCollection
+from nplinker.strains import Strain
+
+
+def test_podp_generate_strain_mappings(monkeypatch, tmp_path):
+    # mock functions called by the tested function
+    mappings_strain_bgc = {
+        "strain1": {"bgc1", "bgc2"},
+        "strain2": {"bgc3"},
+    }
+    mappings_strain_spectrum = {
+        "strain1": {"spec1", "spec2"},
+        "strain2": {"spec3"}
+    }
+
+    monkeypatch.setattr(
+        "nplinker.pairedomics.strain_mappings_generator.extract_mappings_strain_id_original_genome_id",
+        lambda *args: {})  # any return value is fine
```

Comment:
I don't get what is going on here, what does `monkeypatch` represent? And how is it then used? (I see only the `.setattr(...)` methods)

Reply:
`monkeypatch` is [a pytest fixture for mocking](https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html). Here is an [example on mocking functions](https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html#monkeypatching-functions). I'm using the same trick in the unit tests to mock functions.

Here we're testing the function `podp_generate_strain_mappings` (A),  and it calls function `get_mappings_strain_id_bgc_id` (B)  which calls functions like `extract_mappings_strain_id_original_genome_id` (C). So the chain of calling is `A -> B -> C`.

In order to run function A, we need to prepare the inputs for the function B and C, which requires a lot of work. 
However the purpose is to test function A, what we really need from B and C is their output and we don't really care if they work well or not (they are tested separately). So we can use some trick to directly provide the output of C to B and the output of B to A without giving the inputs. This kind of trick is called mocking.